tvbwidgets/exporters/model_exporters.py

```diff
@@ -0,0 +1,191 @@
+            return prop.tolist()
+        return prop.__class__.__name__
+
+
+class JSONModelExporter(ABCModelExporter):
+    """
+    Class used to export a model configuration to json format
+    """
+    file_name = 'model_config.json'
+
+    def __init__(self, model_instance, keys):
+        # type: (Model, list[str]) -> None
+        super(JSONModelExporter, self).__init__(model_instance, keys)
+
+    @property
+    def default_config_name(self):
+        return 'default_config_key'
+
+    def do_export(self):
+        """
+        exports model params and model name as json under the key config_name
+        if that is different to the default key or the default key plus length of the existing keys
+        in the json file + 1
+        """
+        config_dict = self._read_existing_config()
+        config_name = self.config_name if self.config_name != self.default_config_name else self.default_config_name + str(len(config_dict.keys()) + 1)
+        config_dict[config_name] = self._prepare_export_values()
+        with open(self.file_name, 'w') as f:
+            f.write(json.dumps(config_dict))
+
+    def _read_existing_config(self):
+        # type: () -> dict
+        """
+        reads a json file and returns the dict representing the json data or an empty dict
+        if the config file or data in it doesn't exist
+        """
+        file_name = self.file_name
+        config = {}
+        try:
+            with open(file_name, 'r+') as config_file:
+                config = json.loads(config_file.read())
+        except (FileNotFoundError, json.JSONDecodeError):
+            pass
+        return config
+
+    def _prepare_export_values(self):
+        # type: () -> dict
+        """
+        returns a json safe dict containing the model params from the exporter keys
+        """
+        values = {}
+        model_dict = self.model_instance.__dict__
+
+        for key in self.keys:
+            if model_dict.get(key, False):
+                values[key] = self.sanitize_property(model_dict[key])
+
+        values['model'] = self.model_instance.__class__.__name__
+
+        return values
+
+
+class PythonCodeExporter(ABCModelExporter):
+    file_name = 'model_instances.py'
+    models_import = 'tvb.simulator.models'
+
+    def __init__(self, model_instance, keys):
+        # type: (Model, list[str]) -> None
+        super(PythonCodeExporter, self).__init__(model_instance, keys)
+
+    @property
+    def default_config_name(self):
+        return 'default config'
+
+    def do_export(self):
+        class_name = self.model_instance.__class__.__name__
+        values = f'# {self.config_name}\n'
+        # assume that if the file exists the imports also exist
+        if not os.path.exists(self.file_name):
+            values += f'import numpy\nfrom {self.models_import} import *\n'
+
+        model_params = self.get_model_params()
+        values += f'model_instance = {class_name}({model_params})\n\n'
```

Comment:
Could we maybe put these strings some place more visible (e.g. constants in the top area) ?